src/Proto.Remote/EndpointReader.cs

```diff
@@ -50,7 +50,10 @@ public class EndpointReader : Remoting.RemotingBase
                     }
                     else
                     {
-                        target.Request(message, sender);
+                        if (envelope.Sender != null)

```

Comment:
This change should not have been in this specific PR, but it is a small optimization in Remote, not allocating a MessageEnvelope if there is no sender... carry on ;-)